x-pack/plugins/files/server/file_service/internal_file_service.ts

```diff
@@ -167,4 +180,47 @@ export class InternalFileService {
     const updateResponse = await this.soClient.update(this.savedObjectType, id, attributes);
     return updateResponse as FileSavedObject;
   }
+
+  public async findFilesJSON({
+    kind,
+    name,
+    extension,
+    mimeType,
+    meta,
+    status,
+    page,
+    perPage,
+  }: FindFileArgs): Promise<FileJSON[]> {
+    const kueryExpressions: string[] = [];
+
+    const addFilters = (fieldName: string, values: string[] = []): void => {
+      if (values.length) {
+        const orExpressions = values
+          .filter(Boolean)
+          .map(
+            (value) => `${this.savedObjectType}.attributes.${fieldName}: ${JSON.stringify(value)}`
```

Comment:
I'm not 100% sure, but we might need to do something about [special characters](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_reserved_characters) here.

Here is an [example escape function](https://github.com/elastic/kibana/blob/main/src/plugins/share/server/url_service/short_urls/util.ts#L13-L15).

Reply:
Some fields we know for sure will never need escaping, like the `Status` field. But, for example, in user provided `Meta` field there could be anything.

Reply:
Ah cool, I was not aware of the reserved character list. That is really useful!

Reply:
I found `nodeBuilder` and `escapeKuery` which I think will do exactly what we want. Will update with that.